autogluon/utils/tabular/ml/models/lr/lr_model.py

```diff
@@ -0,0 +1,242 @@
+import logging
+import re
+
+import numpy as np
+from pandas import DataFrame
+from sklearn.feature_extraction.text import TfidfVectorizer
+from sklearn.impute import SimpleImputer
+from sklearn.linear_model import LogisticRegression, Ridge, Lasso
+from sklearn.pipeline import FeatureUnion, Pipeline
+from sklearn.preprocessing import StandardScaler, QuantileTransformer
+
+from autogluon.utils.tabular.ml.constants import BINARY, MULTICLASS, REGRESSION
+from autogluon.utils.tabular.ml.models.abstract.abstract_model import AbstractModel
+from autogluon.utils.tabular.ml.models.lr.lr_preprocessing_utils import NlpDataPreprocessor, OheFeaturesGenerator, NumericDataPreprocessor
+
+logger = logging.getLogger(__name__)
+
+
+class AbstractLinearModel(AbstractModel):
+
+    def __init__(self, path: str, name: str, problem_type: str, objective_func, num_classes=None, hyperparameters=None, features=None,
+                 feature_types_metadata=None, debug=0, regression_option='ridge', **kwargs):
+        super().__init__(path=path, name=name, problem_type=problem_type, objective_func=objective_func, hyperparameters=hyperparameters, features=features,
+                         feature_types_metadata=feature_types_metadata, debug=debug)
+        self.types_of_features = None
+        self.regression_option = regression_option
+        self.pipeline = None
+
+        if self.problem_type == REGRESSION:
+            self._get_regression_model(kwargs)
+        else:
+            self._model_type = LogisticRegression
+
+        self.model_params = None
+        self.set_default_params()
+
+    def _get_regression_model(self):
```

Comment:
This can't accept kwargs currently, but it is called with kwargs as a parameter in __init__